### PR TITLE
Improve authorization unit test coverage

### PR DESCRIPTION
**Pull request by shaynafinocchiaro**
# Description
PR to improve authorization unit test coverage. 

Additional minor enhancements:
- Adds a new PR action to run module unit tests
- Updates actions/checkout to v3 and actions/setup-go to v4
- Adds a module unit test make target
- Fixes `TestCsmUpgradePathInvalid` controller unit test to use an invalid config version

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| https://github.com/dell/csm/issues/776|

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have maintained backward compatibility

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [x] Ran module unit tests to verify coverage > 90% and ran existing controller unit tests

### CODE REVIEW

File: controllers/csm_controller_test.go [hunk 1]
```diff
@@ -364,7 +365,7 @@ func (suite *CSMControllerTestSuite) TestCsmUpgradePathInvalid() {
 		annotations = make(map[string]string)
 	}
 	if _, ok := annotations[configVersionKey]; !ok {
-		annotations[configVersionKey] = jumpUpgradeConfigVersion
+		annotations[configVersionKey] = invalidConfigVersion
```
Comment (jooseppi-luna) [inline controllers/csm_controller_test.go:368]: we can revert this change because the unit tests are fixed now

**Review by jooseppi-luna — CHANGES_REQUESTED** [review]
looks great bonus points for improving comments. Undo the change in csm_controllers_test.go and I can approve

File: controllers/csm_controller_test.go (hunk 1 above)
Comment (jooseppi-luna) [reply controllers/csm_controller_test.go:368]: reverting that change should fix the coverage as well, I think?

Comment (shaynafinocchiaro) [reply controllers/csm_controller_test.go:368]: the jumpUpgradeConfigVersion is being used in 2 different tests right now https://github.com/dell/csm-operator/blob/dc1358b4bc3997ca93bf5a9abaee104752419bc0/controllers/csm_controller_test.go#L341

**Review by jooseppi-luna — APPROVED** [review]

**Review by jooseppi-luna — APPROVED** [review]

**Review by xuluna — APPROVED** [review]

**Review by atye — APPROVED** [review]
